core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -562,6 +562,26 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  case class LeaderNode(leaderId: Int, leaderEpoch: Int, node: Node)
+
+  private def getCurrentLeader(tp: TopicPartition, ln: ListenerName): LeaderNode = {
+    val partitionInfoOrError = replicaManager.getPartitionOrError(tp)
```

Comment:
Do we know how this error will be handled? I think it will probably cause the whole request to fail unless we catch it

Reply:
There shouldn't be any error thrown here, `getPartitionOrError` returns `Either[Errors, Partition]`, the error would be handled in the case statement below.

Reply:
Ah right 🤦‍♀️ I was thinking about a different method that throws an error.
So in the case where we have errors, here we just don't return anything in the new fields we are adding?
Do we have a test for this?

Reply:
We should return the default values (-1) if we have errors looking up the new leader, the client should do a full metadata refresh in that case to get the new leader.
I added a test case for failing to look up the new leader in ReplicaManager. It looks more difficult to add a test case for failing to look up the new leader in the metadata cache, since it needs to be in there initially to start processing the request. I can look more into it if you think we need it.

Reply:
I think it would be useful to have to see that the -1 -1 is handled as expected. Maybe you can mock the metadata cache as well? Change it to only return the data once?

Reply:
I added a test for failing to lookup in the metadata cache, it was actually pretty easy with mocking. The error cases should all be covered now.